File: Ozone.WebApi/Ozone.Application/DTOs/Security/UserAuditModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Ozone.Application.DTOs
{
   public class UserAuditModel
    {
        public long Id { get; set; }
        public long? UserId { get; set; }
        public long? StandardId { get; set; }
        public string StandardName { get; set; }
        public string Organization { get; set; }
       
        public long? EacodeId { get; set; }
        public string EacodeName { get; set; }
        public long? NaceCodeId { get; set; }
        public string NacecodeName{ get; set; }
        public long? AuditTypeId { get; set; }
        public string AuditTypeName { get; set; }
        public decimal? Duration { get; set; }
        
        public int? Year { get; set; }
        public long? CertificationBodyId { get; set; }
        public string CertificationBodyName { get; set; }
        public bool? IsDeleted { get; set; }
        public long? ApprovedBy { get; set; }
       
        public DateTime? ApprovedDate { get; set; }
        public long? CreatedBy { get; set; }
        
        public DateTime? CreatedDate { get; set; }
        public long? LastModifiedBy { get; set; }
     
        public DateTime? LastModifiedDate { get; set; }
        public string AuditLevel { get; set; }
        

    }
}

File: angular/src/shared/Dto/user-audit-model.ts
export class UserAuditModel implements IUserAuditModel {
   
    Id: number | undefined
    UserId: number | undefined;
    Organization: string | undefined;
    StandardId: number | undefined;
    Duration: number | undefined;
    Year: number | undefined;
    EacodeId: number | undefined;
    NaceCodeId: number| undefined;
    AuditTypeId: number| undefined;
    ApprovedBy : number| undefined;
    CreatedBy: number| undefined;
    CertificationBodyId: number| undefined;
    LastModifiedBy: number| undefined;
    AuditLevel: string | undefined;
    //SmetaAuditId: number | undefined;

    constructor(data?: UserAuditModel) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }
    code: string;
    name: string;
    id: number;

    init(data?: any) {
        if (data) {
            this.Id = data["Id"];
            this.UserId = data["UserId"];
            this.Organization = data["Organization"];
            this.StandardId = data["StandardId"];
            this.Duration = data["Duration"];
            this.Year = data["Year"];
            this.EacodeId = data["EacodeId"];
            this.NaceCodeId = data["NaceCodeId"];
            this.ApprovedBy = data["ApprovedBy"];
            this.CreatedBy = data["CreatedBy"];
            this.AuditTypeId = data["AuditTypeId"];
            this.CertificationBodyId = data["CertificationBodyId"];
            this.LastModifiedBy = data["LastModifiedBy"];
            this.AuditLevel = data["AuditLevel"];
            //this.SmetaAuditId = data["SmetaAuditId"];
            
        }
    }

    static fromJS(data: any): UserAuditModel {
        data = typeof data === 'object' ? data : {};
        let result = new UserAuditModel();
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["Id"] = this.Id;
        data["UserId"] = this.UserId;
        data["Organization"] = this.Organization;
        data["StandardId"] = this.StandardId;
        data["Duration"] = this.Duration;
        data["Year"] = this.Year;
        data["EacodeId"] = this.EacodeId;
        data["NaceCodeId"] = this.NaceCodeId;
        data["ApprovedBy"] = this.ApprovedBy;
        data["CreatedBy"] = this.CreatedBy;
        data["AuditTypeId"] = this.AuditTypeId;
        data["CertificationBodyId"] = this.CertificationBodyId;
        data["LastModifiedBy"] = this.LastModifiedBy;
        data["AuditLevel"] = this.AuditLevel;
        //data["SmetaAuditId"] = this.SmetaAuditId;

        return data; 
    }

    clone(): UserAuditModel {
        const json = this.toJSON();
        let result = new UserAuditModel();
        result.init(json);
        return result;
    }
}


export interface IUserAuditModel {
    Id: number | undefined
    UserId: number | undefined;
    Organization: string | undefined;
    StandardId: number | undefined;
    Duration: number | undefined;
    Year: number | undefined;
    EacodeId: number | undefined;
    NaceCodeId: number| undefined;
    ApprovedBy : number| undefined;
    CreatedBy: number| undefined;
    AuditTypeId: number| undefined;
    CertificationBodyId: number| undefined;
    LastModifiedBy: number| undefined;
    AuditLevel: string | undefined;
    //SmetaAuditId: number| undefined;
}